estate/models/estate_property.py

```diff
@@ -0,0 +1,117 @@
+from odoo import api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools import float_compare
+
+
+class EstateProperty(models.Model):
+    _name = 'estate.property'
+    _description = 'Property'
+
+    name = fields.Char('Title', required=True)
+    description = fields.Text('Description')
+    postcode = fields.Char('Postcode')
+    date_availability = fields.Date(
+        'Available From',
+        copy=False,
+        default=fields.Date.add(fields.Date.today(), months=+3),
+    )
+    expected_price = fields.Float('Expected Price', required=True)
+    selling_price = fields.Float('Selling Price', readonly=True, copy=False)
+    bedrooms = fields.Integer('Bedrooms', default=2)
+    living_area = fields.Integer('Living Area (sqm)')
+    facades = fields.Integer('Facades')
+    garage = fields.Boolean('Garage')
+    garden = fields.Boolean('Garden')
+    garden_area = fields.Integer('Garden Area (sqm)')
+    garden_orientation = fields.Selection(
+        string='Garden Orientation',
+        selection=[
+            ('north', 'North'),
+            ('south', 'South'),
+            ('east', 'East'),
+            ('west', 'West'),
+        ],
+        default='north',
+    )
+    active = fields.Boolean('Active', default=True)
+    state = fields.Selection(
+        string='State',
+        selection=[
+            ('new', 'New'),
+            ('offer_received', 'Offer Received'),
+            ('offer_accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('canceled', 'Canceled'),
+        ],
+        required=True,
+        copy=False,
+        default='new',
+    )
+    property_type_id = fields.Many2one('estate.property.type', string='Property Types')
+    property_tag_ids = fields.Many2many('estate.property.tag', string='Property Tags')
+    buyer = fields.Many2one('res.partner', string='Buyer', copy=False)
+    salesperson = fields.Many2one(
+        'res.users', string='Salesperson', default=lambda self: self.env.user
+    )
+    offer_ids = fields.One2many(
+        'estate.property.offer', 'property_id', string='Property Offers'
+    )
+    total_area = fields.Integer('Total Area (sqm)', compute='_compute_total_area')
+    best_price = fields.Float('Best Offer', compute='_compute_best_price')
+
+    _check_expected_price = models.Constraint(
+        'CHECK(expected_price > 0)', 'The expected price must be strictly positive.'
+    )
+
+    @api.depends('garden_area', 'living_area')
+    def _compute_total_area(self):
+        for property in self:
+            property.total_area = property.garden_area + property.living_area
+
+    @api.depends('offer_ids.price')
+    def _compute_best_price(self):
+        for property in self:
+            if len(self.offer_ids):
```

Comment:
```suggestion
            if self.offer_ids:
```
Yours is completely okay and correct but you don't have to call `len()`